electrumsv/wallet.py

```diff
@@ -1628,8 +1596,9 @@ def update_key_activation_state(self, keyinstances: List[KeyInstanceRow], activa
             if activate:
                 new_flags = cast(KeyInstanceFlag, old_flags | KeyInstanceFlag.IS_ACTIVE)
             else:
-                # will not deactivate if USER_SET_ACTIVE flag is set
-                new_flags = cast(KeyInstanceFlag, old_flags & KeyInstanceFlag.INACTIVE_MASK)
+                # if USER_SET_ACTIVE flag is set - this flag will remain
+                new_flags = cast(KeyInstanceFlag, old_flags & (KeyInstanceFlag.INACTIVE_MASK |
+                    KeyInstanceFlag.USER_SET_ACTIVE))
```

Comment:
Can you double check that I have this right? 
`old_flags & (KeyInstanceFlag.INACTIVE_MASK | KeyInstanceFlag.USER_SET_ACTIVE)`

I want to wipe the `IS_ACTIVE` flag but leave the `USER_SET_ACTIVE` (if it has been set)

Reply:
The casts should not be in your logic, but in the given original table read before it returns the results. Otherwise it should be fine AFAICT.